### DISCUSSION TITLE
Ignoring noqa lines

### DISCUSSION THREAD

**Discussion by bnomis** [astral-sh/ruff#4703] [Q&A] [answered]
When writing scripts to interact with django it's quite common to import django and call it's setup function in one line. Typically the top of a file looks like this:

```python
#!/usr/bin/env python
import argparse
import sys
import time

import django; django.setup()  # noqa
from django.conf import settings
```

Ruff reports a "Module level import not at top of file Ruff (E402)" for the line `from django.conf import settings` and any following import lines because of the line `import django; django.setup()  # noqa`. I was hoping the `# noqa` at the end of the line would cause the line to, essentially, be ignored but it seems not. Is there a way to tell Ruff that it's OK to ignore the line completely?

Thanks

**Comment by charliermarsh** [comment, marked as answer]
Unfortunately no, there isn't a way to tell Ruff to ignore a line for the purpose of determining the "start-of-file" import boundary.

The best you can do, I think, is turn off `E402` for that module, like:

```py
import argparse
import sys
import time

# ruff: noqa: E402
import django; django.setup()
from django.conf import settings
```

**Reply by charliermarsh** [reply]
(Something to consider when we revisit suppression comments more broadly.)
